pyproject.toml

```diff
@@ -1,3 +1,47 @@
+[tool.poetry]
+name = "pygmt"
+version = "0.3.1"
```

Comment:
> Just in case you don't notice it: poetry doesn't work well with setuptools_scm, so the version is `v0.3.1`, not `v0.3.1.dev100+xxxxx`.

Yeah I realize that this hardcoding is bad. There's https://github.com/mtkennerly/poetry-dynamic-versioning but doesn't look like a very official solution.

To be honest, this PR isn't intended to replace `setup.py` just yet. Just wanted to see if mixing `conda` and `poetry` might shave a few minutes off the CI test time.